x-pack/legacy/plugins/lens/public/indexpattern_plugin/layerpanel.tsx

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import _ from 'lodash';
+import React, { useState, useMemo } from 'react';
+import {
+  EuiComboBox,
+  // @ts-ignore
+  EuiHighlight,
+  EuiButtonEmpty,
+  EuiIcon,
+  EuiIconTip,
+  EuiFlexGroup,
+  EuiFlexItem,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { I18nProvider } from '@kbn/i18n/react';
+import { DatasourceLayerPanelProps } from '../types';
+import { IndexPatternPrivateState } from './indexpattern';
+import { isLayerTransferable, updateLayerIndexPattern } from './state_helpers';
+
+export interface IndexPatternLayerPanelProps extends DatasourceLayerPanelProps {
+  state: IndexPatternPrivateState;
+  setState: (newState: IndexPatternPrivateState) => void;
+}
+
+export function LayerPanel({ state, setState, layerId }: IndexPatternLayerPanelProps) {
+  const [isChooserOpen, setChooserOpen] = useState(false);
+  const indexPatterns = useMemo(
+    () =>
+      Object.values(state.indexPatterns)
+        .filter(indexPattern => indexPattern.id !== state.layers[layerId].indexPatternId)
+        .map(indexPattern => ({
+          ...indexPattern,
+          isTransferable: isLayerTransferable(state.layers[layerId], indexPattern),
+        })),
+    [state.indexPatterns, layerId, state.layers[layerId]]
+  );
+  const currentIndexPatternId = state.layers[layerId].indexPatternId;
+
+  return (
+    <I18nProvider>
+      <EuiFlexGroup justifyContent="flexEnd">
+        {isChooserOpen ? (
+          <EuiFlexItem>
+            <EuiComboBox
```

Comment:
Not sure if we want to fix this kind of thing in these PRs, but the styling here is a bit weird, as the combo box and text overflows its container.

Reply:
The styling is rudimentary and not supposed to stay like that for release, it's going to get polished with the rest of the config panel.